### ISSUE TITLE
rtl fix, utf8 feed fetch problem & identi.ca support

### ISSUE THREAD

**Issue by GoogleCodeExporter** [KGIII/pino-twitter#36]
```
hi there,

i have created a patch witch has following contents:

 1. there's a problem with parsing feeds witch has utf8 characters. this
problem causes an error like this:

----------
Entity: line 789: parser error : Couldn't find end of Start Tag
profile_text_ line 789
   <profile_text_
                 ^
Entity: line 789: parser error : Premature end of data in tag user line 778
   <profile_text_
                 ^
Entity: line 789: parser error : Premature end of data in tag status line 767
   <profile_text_
                 ^
Entity: line 789: parser error : Premature end of data in tag statuses line 2
   <profile_text_
                 ^
Segmentation fault (core dumped)
-----------

& problem is you're using string.length instead of string.size() when you
use Parser.parse_memory.

 2. i have added jquery & one extension witch automatically detects rtl
tweets/dents & sets RTL direction for them.

 3. i have changed feed urls to identi.ca (i'm not using twitter). if you
don't want to support identi.ca at this time just ignore it
```

Original issue reported on code.google.com by `everpl...@gmail.com` on 9 Jan 2010 at 4:15

Attachments:
- [pino.patch](https://storage.googleapis.com/google-code-attachments/pino-twitter/issue-36/comment-0/pino.patch)

**Comment by GoogleCodeExporter** [comment]
```
Hi.
Thank's for this patch. I've already used your changes in the parser. And I 
will think 
about identi.ca support (this is simpler than I thought). But I don't want to 
use JS in 
templates at this moment.
```

Original comment by `tro...@gmail.com` on 9 Jan 2010 at 5:47
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
hmm, ok. but we really need to implement direction detection per tweet/dent. 
take a
look at attachment. first dent is in english but others are in persian (written 
from
right to left) as you see pino detects it & automatically sets rtl direction &
alignment. in my experience this js files didn't effect on performance of pino.

if you still don't want to use js, we can simply implement it via vala & add 
new css
rule.
```

Original comment by `everpl...@gmail.com` on 9 Jan 2010 at 6:12

Attachments:
- [Screenshot-pino.png](https://storage.googleapis.com/google-code-attachments/pino-twitter/issue-36/comment-2/Screenshot-pino.png)

**Comment by GoogleCodeExporter** [comment]
```
> if you still don't want to use js, we can simply implement it via vala & add 
new css
rule.

That's nice idea.
```

Original comment by `tro...@gmail.com` on 9 Jan 2010 at 6:22

**Comment by GoogleCodeExporter** [comment]
```
i have implemented js direction detector via vala. it's really simpler than JS 
one :)
```

Original comment by `everpl...@gmail.com` on 9 Jan 2010 at 8:35

Attachments:
- [Screenshot-pino.png](https://storage.googleapis.com/google-code-attachments/pino-twitter/issue-36/comment-4/Screenshot-pino.png)
- [pino-rtl.patch](https://storage.googleapis.com/google-code-attachments/pino-twitter/issue-36/comment-4/pino-rtl.patch)

**Comment by GoogleCodeExporter** [comment]
```
Great work, thanks a lot! I will add your patch to the 0.1.3 version, if you 
don't mind.
And I want to add you to the authors list. Can you give me your name and email?
```

Original comment by `tro...@gmail.com` on 9 Jan 2010 at 10:58
- Changed state: **Accepted**
- Added labels: **Priority-High**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
good job :)

my name is behrooz shabani & my email is everplays [at] gmail (dot) com

ah, i'm in iran & google code is blocked here by google, i have to use vpn (or 
such
tools) to access it & unfortunately vpn is not accessible here all time as 
government
blocks it but i will try to do some works on pino because i really like it.
```

Original comment by `everpl...@gmail.com` on 10 Jan 2010 at 7:48

**Comment by GoogleCodeExporter** [comment]
```
Fixed by f780c6a625 revision.
```

Original comment by `tro...@gmail.com` on 11 Jan 2010 at 5:51
- Changed state: **Fixed**
